service/tablet_allocator.cc

```diff
@@ -1120,20 +1121,32 @@ class load_balancer {
 
             // Drop targets which would increase max rack load.
 
-            max_rack_load = std::max_element(rack_load.begin(), rack_load.end(),
-                                                  [] (auto& a, auto& b) { return a.second < b.second; })->second;
+            int min_rack_load = std::numeric_limits<int>::min();
+            int max_rack_load = std::numeric_limits<int>::max();
+            for (const auto& [_, load] : rack_load) {
+                if (min_rack_load < load) {
+                    min_rack_load = load;
+                }
+                if (max_rack_load > load) {
+                    max_rack_load = load;
+                }
+            }
 
+          // FIXME: indentation
+          if (min_rack_load < max_rack_load) {
```

Comment:
This will allow breaking constraints not only during decommission, but any migration.